OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitNavigationActivity.java

```diff
@@ -245,17 +297,24 @@ private void updateDataModel() {
             tryToUpdateForEndOfSequential();
         }
 
-        if (pagerAdapter != null)
+        if (pagerAdapter != null) {
             pagerAdapter.notifyDataSetChanged();
+        }
+
+        // Initialize indicator layout for first opened unit
+        indicatorController.initialize(courseUnitBlockCounterMap.get(selectedUnit.getParent().getId()).size());
```

Comment:
:warning: There is no need of `courseUnitBlockCounterMap`, you can recalculate the required indicator sized by using `isVideoMode` flag.